### PR TITLE
#12461 View existing and create new email templates in the configurat…

### LINKED ISSUE #12461: View existing and create new email templates in the configuration screen [5]
### Feature Description

Add an email templates view that allows users to create new email templates that can then be used to send external emails, e.g. to case or contact persons (see #12402).

A lot of the logic can be taken over from document templates.

### Added Value/Benefit

Having set up email templates will allow users to send emails a lot faster and with less manual effort.

### Acceptance Criteria

- [x] A new `EXTERNAL_EMAILS` feature type has been added (added to the feature extensions section of the enum), disabled by default
- [x] A new "Email templates" view has been added to the Configurations page, next to document templates
- [x] The view contains sections for different entities: "Case email templates", "Contact email templates", "Event participant email templates" and "Travel entry email templates"
- [x] An "Upload template" button next to each section brings up a popup window similar to the one for document templates
- [ ] An email template guide is provided (@MateStrysewske can take care of this)
- [x] An example template can be downloaded; content can be identical to the one for document templates, but with updated heading; example needs to be provided as .txt instead of .docx
- [x] Users can use the popup to upload new templates as .txt files (other file types are not supported)
- [x] Only users with the `EMAIL_TEMPLATE_MANAGEMENT` user right introduced in #12466 can see and access the new view, and the new feature type has to be enabled
- [x] Email templates work exactly as document templates, e.g. they support the same variables
- [x] The templates are stored in a new folder called `emailtemplates` in the /opt/sormas/custom folder; templates for the respective entities will be stored in subfolders named after the entity (i.e. `cases`, `contacts`, etc.)
- [x] Clean and maintainable code is provided (i.e. the document template code is not simply copied over for email templates, but ideally there's a lot of joint code as both work very similarly; there might, however, be differences in the future, so they can't use the same DTO)

### Mockups

image

image

### PR DESCRIPTION
**Pull request by leventegal-she**
…ion screen

Fixes #12461

@MateStrysewske Plase add the user guide. The file name used in the code is `SORMAS_Email_Template_Guide.pdf`

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12762

File: sormas-api/src/main/java/de/symeda/sormas/api/feature/FeatureType.java [hunk 1]
```diff
@@ -331,7 +331,8 @@ public enum FeatureType {
 			CASE_SURVEILANCE,
 			CONTACT_TRACING },
 		null,
-		ImmutableMap.of(FeatureTypeProperty.S2S_SHARING, Boolean.FALSE));
+		ImmutableMap.of(FeatureTypeProperty.S2S_SHARING, Boolean.FALSE)),
+	EXTERNAL_EMAILS(true, false, null, null, null);
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/feature/FeatureType.java:335]: Please put this into the correct section (see issue description)

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
Looks good apart from this minor thing; I'll take care of the guide tomorrow

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12762

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**